projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx

```diff
@@ -71,18 +73,26 @@ export default function ProductInterstitial( {
 	directCheckout = false,
 	highlightLastFeature = false,
 	ctaCallback = null,
+	feature = null,
 } ) {
 	const { detail } = useProduct( slug );
 	const { detail: bundleDetail } = useProduct( bundle );
 	const { activate, isPending: isActivating } = useActivate( slug );
 
+	// Get the post activation URL for the product.
+	let redirectUri = detail?.postActivationUrl || null;
+	// If the interstitial is highlighting a specific feature, use the post checkout URL for that feature, if available.
+	if ( feature && detail?.postActivationUrlsByFeature?.[ feature ] ) {
+		redirectUri = detail.postActivationUrlsByFeature[ feature ];
+	}
```

Comment:
I think all of this could be shortened into a one line ternary like this
```jsx
const redirectUri = feature && detail?.postActivationUrlsByFeature?.[ feature ] ? detail?.postActivationUrlsByFeature?.[ feature ] : detail?.postActivationUrl;
```

But that isn't blocking and is just my opinion 😄